pkg/node/store/store.go

```diff
@@ -39,17 +40,23 @@ var (
 
 // NodeObserver implements the store.Observer interface and delegates update
 // and deletion events to the node object itself.
-type NodeObserver struct{}
+type NodeObserver struct {
+	manager *nodemanager.Manager
+}
 
 func (o *NodeObserver) OnUpdate(k store.Key) {
 	if n, ok := k.(*node.Node); ok {
-		n.OnUpdate()
+		nodeCopy := *n
+		nodeCopy.Source = node.FromKVStore
+		o.manager.NodeUpdated(nodeCopy)
```

Comment:
Be aware this is not a copy, the internal slices of `n` were not copied. Please use DeepCopy